### PR TITLE
QUANT-4356 make json compatible with ruby 3

### PR DESCRIPTION
**Pull request by nbhattml**
Output of 
`gem build json-java.gemspec`

```
 Successfully built RubyGem
  Name: json
  Version: 1.8.6.1
  File: json-1.8.6.1-java.gem
```

### CODE REVIEW

File: lib/json/common.rb [hunk 1]
```diff
@@ -152,7 +152,7 @@ class MissingUnicodeSupport < JSONError; end
   # * *object_class*: Defaults to Hash
   # * *array_class*: Defaults to Array
   def parse(source, opts = {})
-    Parser.new(source, opts).parse
+    Parser.new(source, **(opts||{})).parse
```
Comment (msharma-ml) [inline lib/json/common.rb]: In params we are already assign the empty hash to opts  `opts = {}` do we need to explictly do `|| {}` ?

File: lib/json/common.rb [hunk 2]
```diff
@@ -175,7 +175,7 @@ def parse!(source, opts = {})
       :max_nesting  => false,
       :allow_nan    => true
     }.update(opts)
-    Parser.new(source, opts).parse
+    Parser.new(source, **(opts||{})).parse
```
Comment (msharma-ml) [inline lib/json/common.rb]: We don't need to explicitly do the `|| {}` as in the method params we are already doing it.

File: lib/json/common.rb (hunk 1 above)
Comment (nbhattml) [reply lib/json/common.rb]: done

**Review by msharma-ml — APPROVED** [review]
LGTM!

**Pull request merged**